### ISSUE TITLE
подключение COLT к проекту node.js+angular.js

### ISSUE THREAD

**Issue by theasmoth** [code-orchestra/colt-ui#5]
sublime 3 + COLT
node.js + angular.js
проект имеет структуру
project
---> static
--- ---> js
--- --- ---> app.js \* angular.js приложение
---> views \* директория шаблонов для angular.js приложения
--- ---> index.ejs \* html шаблон отдаваемый сервером
---> index.js \* node.js start

многие директории и файлы опущены (images, css и другие шаблоны)

как в таком проекте запустить COLT для front-end разработки?
пытался по разному запустить, курил документацию, но так и не понял

**Comment by potapenko** [comment]
откройте в sublime рутовую папку проекта 
откройте главный документ
вызывайте команду "Run With COLT" - (первый раз спросит внизу окна путь к COLT)

**Comment by potapenko** [comment]
а, смешаный проект…. и node  и ангулар
через live-proxy режим пробовали запускать?

**Comment by potapenko** [comment]
angular приложение  без node совсем не живет?

**Comment by theasmoth** [comment]
главный документ это node приложение? angular приложение? или главный шаблон страницы?
на шаблоне кнопки Run with COLT неактивны видимо из-за расширения отличного от *.html

через live-proxy не пробовал...
для этого node-webkit поставить надо?

angular без node не живет совсем у нас

**Comment by potapenko** [comment]
да, в вашем случае нужно запускать в режиме live-proxy. 
тут нужно выбирать, над чем вы будете работать. 
если над html - то нужен режим live-proxy. 
это http watcher. запустите в обычном режиме (без COLT) сервер node.js
сетевой путь к главному документу пропишите в main, например, http://localhost:8080/index.html https://www.dropbox.com/s/sjrszdedtz9nmi5/%D0%A1%D0%BA%D1%80%D0%B8%D0%BD%D1%88%D0%BE%D1%82%202014-04-30%2013.27.29.png

Запустите сессию - нажмите зеленую молнию в COLT.
при изменении файлов локально, должны видны изменения на запущенной в браузере странице 

Чтобы запускать из sublime c этим адресом, добавьте meta тэг в index.html - 

```
<meta name="colt-main-document" content="http://localhost:8080/index.html">
```

и выберете "Run With COLT", когда отрыт этот документ.

**Comment by theasmoth** [comment]
спасибо, попробую
по результатам отпишусь

**Comment by potapenko** [comment]
пока не придумали как такие смешанные проекты запускать просто. 
чтобы и нода и html частично перегружались при одном запуске. 
может у вас будут идеи?

**Comment by potapenko** [comment]
пока нужно запускать два COLT - один для node, другой для статики через live-proxy

**Comment by potapenko** [comment]
то что node была заблокирована - нужно было прописать путь к node в настройках phpstorm, это наша недоработка - будем исправлять.

**Comment by theasmoth** [comment]
завтра на работе протестирую
мне нужен COLT для angular'a, COLT для бэкэнда не требуется

еще вопрос, будет ли работать автокомплит для {{}} в файлах другого расширения (не html)?

**Comment by potapenko** [comment]
jade? или другой шаблонизатор? мы можем поддержать, но для этого нам нужно понять какой html из этого шаблонизатора будет создаваться. Нам нужно понять какой $scope в данной точке в запущенном приложении. если были бы source-maps - это бы нам помогло. 
опишите кейс? можем подумать.

**Comment by potapenko** [comment]
или имеется ввиду template у шаблона? в директиве?
кстати если внешний темлейт - html - то в таком файле мы должны автокомлитить в {{}}

**Comment by theasmoth** [comment]
да, есть несколько шаблонов для директив и роутов, у них имеется расширение *.ejs, осталось по привычке от бывших разработчиков
т.е. я так понимаю не имеет значение какой файл будет, главное что он прописан в templateUrl директивы/роута

**Comment by potapenko** [comment]
мы пока не автокомплитим в файлах в расширении отличном от хтмл, 
вынесем этот конфиг в настройки sublime. поставил задачу. несколько дней - будет update  пакета.

**Comment by potapenko** [comment]
Кстати мы можем не расширение смотреть, а highliter sublime, подсветка для html выбрана - попробуем...

**Comment by theasmoth** [comment]
попробовал запустить в режиме live-proxy, выяснилось следующее:
1. сессии не запускаются если прописан URL http://localhost/ (http://localhost:80/)
2. Run with COLT неактивен из нашего главного шаблона т.к. у него другое расширение

если в main ввести http://localhost:80
1. вылетает окошко с надписью "No message" и кнопкой "Close", при щелчке по которой это окно не закрывается, закрыть можно щелчком по крестику справа сверху

если в main ввести http://localhost:80/ ("/" в конце добавилось)
1. появляется окошко как и предыдущее
2. появляется окошко "The URL specified is invalid: '' + url"
3. при этом открывается окно в браузере с моим IP и портом 9091:
HTTP ERROR 404
Problem accessing /. Reason:
Not Found
Powered by Jetty://

а еще наверное в логе проекта не помешала бы кнопка очистить или время напротив каждой строчки, чтобы можно было видеть с какой сессией проблема

**Comment by vasa-chi** [comment]
Подтверждаю такое поведение. В моем случае нода отдаёт статику и пробрасывает апи, разрабатываемое другим разработчиком. Есть ли возможность пробросить HTTP-запросы через кольт к ноде (по маске например, или в случае, если запрашиваемый юрл не найден)?

**Comment by knoff** [comment]
Аналогично и у меня. Попытка доступа к любому ресурсу в режиме Live-proxy выдает HTTP Error 404. Окна с ошибками то появляются, то нет (пока не удалось выявить закономерность), а вот 404 ошибка в браузере стабильно.
